p2p/tools/local_network.py

```diff
@@ -0,0 +1,378 @@
+import asyncio
+import os
+import random
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    NamedTuple,
+    List,
+    Tuple,
+)
+
+import pytest
+
+from p2p.cancel_token import CancelToken, wait_with_token
+from p2p.exceptions import OperationCancelled
+from p2p.service import BaseService
```

Comment:
You mentioned wanting to pull this module to its own pypi distribution. I assume that means removing all these p2p dependencies. Any ideas/plans for that?

Reply:
Yes, cancel token is going to get it's own pypy distribution `asyncio-cancel-token, and then the new library which I'm calling `pytest-asyncio-networking-simulator` will depend on that distribution.